docs/documentation/server_admin/topics/clients/oidc/con-token-role-mappings.adoc

```diff
@@ -0,0 +1,80 @@
+in the tab *Service accounts roles* of the particular client.
+
+== Role protocol mappers
+
+Similarly like the other claims, the roles are added to the access token issued for the client by the dedicated <<_protocol-mappers,Protocol mappers>>. There is a <<_client_scopes_protocol,Built-in client scope *roles*>>
+defined in the realm. Since it is a <<proc_updating_client_scopes_{context},Realm default client scope>>, it is defined by default as a <<_client_scopes_linking,Default client scope>> for every realm client.
+You can see this client scope in the admin console by looking at the tab *Client scopes* and then looking for the *roles* client scope. This client scope contains these protocol mappers by default:
+
+* The protocol mapper *realm roles* - The protocol mapper is used to add the realm roles to the token claim. By default configuration looks like this:
+
+.Realm roles mapper
+image:images/mapper-oidc-realm-roles.png[]
+
+* The protocol mapper *client roles* - The protocol mapper is used to add the client roles to the token claim. By default, configuration looks like this:
+
+.Client roles mapper
+image:images/mapper-oidc-client-roles.png[]
+
+* The protocol mapper *audience resolve* - The protocol mapper is used to fill the `aud` claim in the access token based on the applied client roles. The details about this mapper are in the <<_audience_resolve,Audience resolve section>>.
+
+As you can see in the configuration of realm roles and client roles mappers, it is possible to configure:
+
+* If roles are added just to the access token or also to other tokens, like for example ID token. By default, roles are added to the access token and to the introspection endpoint.
+
+* What are the claims where the roles would be added. By default, the realm roles are added to the `realm_access` claim. So for example the claim in the JWT token containing 2 realm roles `role1` and `role2` will look similar to this:
++
+[source,json]
+----
+"realm_access": {
+  "roles": [ "role1", "role2" ]
+}
+----
++
+The client roles are added to the `resource_access` token claim by default. This claim will look like this in the token, which contains
+client roles `manage-account` and `manage-account-links` of client `account` and client role `target-client1-role` of the client `target-client1`:
++
+[source,json]
+----
+"resource_access": {
+  "target-client1": {
+    "roles": [ "target-client1-role" ]
+  },
+  "account": {
+    "roles": [ "manage-account", "manage-account-links" ]
+  }
+}
+----
+
+By adjusting the configuration option *Token claim name* of the role protocol mappers, it is possible to specify that these roles would be added to the token in the configured claim.
+
+If you want to update the role claims just for one specific client (For example client `foo` expects the realm roles in the claim `my-realm-roles` instead of the claim `realm_access`), then it is
```

Comment:
```suggestion
If you want to update the role claims just for one specific client (For example, client `foo` expects the realm roles in the claim `my-realm-roles` instead of the claim `realm_access`), then it is
```